### PR TITLE
feat(tests): add a method to generate the IRI from a resource

### PR DESCRIPTION
**Pull request by GaryPEGEOT**
| Q             | A
| ------------- | ---
| Branch?       | main
| Tickets       |
| License       | MIT
| Doc PR        | 

Use case: 

```php
$code = CouponCodeFactory::createOne(['code' => 'FOOBAR_123']);

$response = $client->request('POST', '/cart', [
    'json' => [
        ...
        'code' => getIriForResource($code->object()), // Instead of findIriBy(['code' => 'FOOBAR_123']) or manually generating IRI                
    ],
]);

self::assertResponseIsSuccessful();

```

### CODE REVIEW

File: src/Symfony/Bundle/Test/ApiTestCase.php [hunk 1]
```diff
@@ -94,9 +94,17 @@ protected function findIriBy(string $resourceClass, array $criteria): ?string
             return null;
         }
 
+        return $this->getIriForResource($item);
+    }
+
+    /**
+     * Generate the IRI of a resource item.
+     */
+    protected function getIriForResource(object $resource): ?string
```
Comment (soyuka) [inline src/Symfony/Bundle/Test/ApiTestCase.php:103]: what about using the same method name as for the iri converter?

Comment (GaryPEGEOT) [reply src/Symfony/Bundle/Test/ApiTestCase.php]: Fine by me!

Comment (soyuka) [reply src/Symfony/Bundle/Test/ApiTestCase.php]: the test needs fixing then :)

Comment (GaryPEGEOT) [reply src/Symfony/Bundle/Test/ApiTestCase.php]: My bad 😅 fixed

**Pull request merged**